parsing/location.ml

```diff
@@ -423,10 +438,8 @@ let () =
       | Sys_error msg ->
           Some (errorf ~loc:(in_file !input_name)
                 "I/O error: %s" msg)
-      | Warnings.Errors n ->
-          Some
-            (errorf ~loc:(in_file !input_name)
-             "Some fatal warnings were triggered (%d occurrences)" n)
+
+      | Warnings.Errors _ -> Some non_displayed_error
```

Comment:
Was about the following: drop the Warnings.Error exception constructor altogether, have Warnings.check_fatal returns a boolean, and explicitly adapt all driver code to behave as they do currently (either exiting the process, or continuing in the case of the toplevel) when fatal warnings have been reported?  It's a much more invasive change, but it will also paves the way to a nicer treatment of reporting multiple errors in general (e.g. adapting the type-checker to report errors without failing necessarily at the first one).